scripts/test-check-false-positives.sh

```diff
@@ -19,54 +19,74 @@ cleanup() {
-        echo "Error: Missing expected errors file: ${EXPECTED_ERRORS_FILE}"
-      fi
-      RESULTS_NO_SPACES="build/test-results-no-spaces.xml"
-      cat build/test-results/*.xml | tr -d '\n' > ${RESULTS_NO_SPACES}
-
-      # check number of expected errors
-      number_errors=$(cat build/test-results/*.xml | grep "<failure>" | wc -l)
-      expected_errors=$(cat ${EXPECTED_ERRORS_FILE} | wc -l)
-
-      if [ ${number_errors} -ne ${expected_errors} ]; then
-          echo "Error: There are unexpected errors in ${PACKAGE_UNDER_TEST}"
-          exit 1
-      fi
-
-      # check whether or not the expected errors exist in the xml files
-      while read -r line; do
-        cat ${RESULTS_NO_SPACES} | grep -E "${line}"
-      done < ${EXPECTED_ERRORS_FILE}
-      rm -f build/test-results/*.xml
-      rm -f ${RESULTS_NO_SPACES}
-      exit 0
-    elif [ $r == 0 ]; then
-      echo "Error: Expected to fail tests, but there was none failing"
+  exit $r
+}
+
+function check_expected_errors() {
+  local package_root=$1
+  local package_name=$(basename $1)
+  local expected_errors_file="${package_root%/}.expected_errors"
+  local result_tests="build/test-results/${package_name}_*.xml"
+  local results_no_spaces="build/test-results-no-spaces.xml"
+
+  if [ ! -f ${expected_errors_file} ]; then
+    echo "No unexpected errors file in ${expected_errors_file}"
+    return
+  fi
+
+  rm -f ${result_tests}
+  (
+    cd $package_root
+    elastic-package test -v --report-format xUnit --report-output file --test-coverage --defer-cleanup 1s || true
+  )
+
+  cat ${result_tests} | tr -d '\n' > ${results_no_spaces}
+
+  # check number of expected errors
+  local number_errors=$(cat ${result_tests} | grep "<failure>" | wc -l)
+  local expected_errors=$(cat ${expected_errors_file} | wc -l)
+
+  if [ ${number_errors} -ne ${expected_errors} ]; then
+      echo "Error: There are unexpected errors in ${package_name}"
       exit 1
-    fi
   fi
 
-  exit $r
+  # check whether or not the expected errors exist in the xml files
+  while read -r line; do
+    cat ${results_no_spaces} | grep -E "${line}"
+  done < ${expected_errors_file}
+
+  rm -f ${result_tests}
+  rm -f ${results_no_spaces}
 }
 
-trap cleanup EXIT
+function check_build_output() {
+  local package_root=$1
+  local expected_build_output="${package_root%/}.build_output"
+  local output_file="$PWD/build/elastic-package-output"
 
-export ELASTIC_PACKAGE_LINKS_FILE_PATH="$(pwd)/scripts/links_table.yml"
+  if [ ! -f ${expected_build_output} ]; then
+    (
+      cd $package_root
+      elastic-package build -v
+    )
+    return
+  fi
 
-OLDPWD=$PWD
-# Build/check packages
-for d in test/packages/${PACKAGE_TEST_TYPE:-false_positives}/${PACKAGE_UNDER_TEST:-*}/; do
   (
-    cd $d
-    elastic-package check -v
+    cd $package_root
+    elastic-package build 2>&1 | tee $output_file || true # Ignore errors here
   )
-done
-cd -
+
+  diff -w -u $expected_build_output $output_file || (
```

Comment:
Probably this needs to be prefixed by $package_root. Since the $output_file is generated after changing to that directory in that `( )` block.

```suggestion
  diff -w -u $expected_build_output $package_root/$output_file || (
```

Reply:
Yes, this is why I was using an absolute path. I have added an `mkdir -p` to ensure that the parent directory exists.

Reply:
Ok, dismiss this suggestion
just show this previous error:

> tee: /opt/buildkite-agent/builds/bk-agent-prod-gcp-1694523471797672347/elastic/elastic-package/build/elastic-package-output: No such file or directory